src/rabbit_queue_consumers.erl

```diff
@@ -179,7 +182,7 @@ erase_ch(ChPid, State = #state{consumers = Consumers}) ->
             All = priority_queue:join(Consumers, BlockedQ),
             ok = erase_ch_record(C),
             Filtered = priority_queue:filter(chan_pred(ChPid, true), All),
-            {[AckTag || {AckTag, _CTag} <- queue:to_list(ChAckTags)],
+            {[AckTag || {AckTag, _CTag} <- maps:to_list(ChAckTags)],
```

Comment:
This has to traverse the entire collection. If we can avoid this, it should make a difference.